java-client/src/main/java/org/opensearch/client/json/JsonDataImpl.java

```diff
@@ -109,6 +109,12 @@ public <T> T deserialize(JsonpDeserializer<T> deserializer, JsonpMapper mapper)
     public void serialize(JsonGenerator generator, JsonpMapper mapper) {
         if (value instanceof JsonValue) {
             generator.write((JsonValue) value);
+        } else if (value instanceof String) {
```

Comment:
:+1: thanks @playLeo , could you please just cover the missing types (Boolean, BigDecimal, Short, Long, Float, ...) thank you.